File: Modules/ServerTypes/Ticket/TicketForm.ts
import { initFormType, LookupEditor, MultipleImageUploadEditor, PrefixedContext, TextAreaEditor } from "@serenity-is/corelib";
import { CommentListEditor } from "../../Ticket/Comment/CommentListEditor";

export interface TicketForm {
    SystemId: LookupEditor;
    ProblemId: LookupEditor;
    Description: TextAreaEditor;
    CommentList: CommentListEditor;
    FilesPath: MultipleImageUploadEditor;
    StatusId: LookupEditor;
    LastActionId: LookupEditor;
}

export class TicketForm extends PrefixedContext {
    static readonly formKey = 'Ticket.Ticket';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TicketForm.init) {
            TicketForm.init = true;

            var w0 = LookupEditor;
            var w1 = TextAreaEditor;
            var w2 = CommentListEditor;
            var w3 = MultipleImageUploadEditor;

            initFormType(TicketForm, [
                'SystemId', w0,
                'ProblemId', w0,
                'Description', w1,
                'CommentList', w2,
                'FilesPath', w3,
                'StatusId', w0,
                'LastActionId', w0
            ]);
        }
    }
}

File: Modules/Ticket/Ticket/TicketForm.cs
namespace Ticketing.Ticket.Forms;

[FormScript("Ticket.Ticket")]
[BasedOnRow(typeof(TicketRow), CheckNames = true)]
public class TicketForm
{
    [Category("Problem")]
    [HalfWidth]
    public int SystemId { get; set; }
    [HalfWidth]
    public int ProblemId { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }

    [Category("Attachments")]
    [CommentListEditor, HalfWidth]
    public List<CommentRow> CommentList { get; set; }
    [HalfWidth]
    public string FilesPath { get; set; }

    [Category("Workflow")]
    [HalfWidth]
    public int StatusId { get; set; }
    [HalfWidth]
    public int LastActionId { get; set; }
}